src/core.rs

```diff
@@ -114,11 +142,34 @@ pub trait BidirectionalCollection: Collection
 where
     Self::Whole: BidirectionalCollection,
 {
+    /// Backsteps the given position to position just before i.
+    ///
+    /// # Precondition
+    ///   - i != start()
+    fn backstep(&self, i: &mut Self::Position);
```

Comment:
is backstep a good name?

Reply:
Only if the other operation is `step` or `forwardstep`.  Are there Rust conventions that relate semantics to names?  The reason Swift uses the `form` prefix for operations like this is so that they don't read like they are mutating the receiver.  `x.backstep(i)` might be read as stepping `x` backward by `i` places.

Reply:
> Are there Rust conventions that relate semantics to names?

Not aware of the one applicable here. Currently what I used is:
`after` for getting next position, `advance` for mutating position to next position.
Similar for `before` and `backstep`.

> `x.backstep(i)` might be read as stepping `x` backward by `i` places.

Totally agree here. I guess then `position_after`, `form_position_after`, `position_before`, `form_position_before` would be better name. They are little bit more explicit but surely not ambiguous.

Reply:
IMO `prior`/`next`/`form_next`/`form_prior` might be better.  If you call the parameters `position` the usage will read well with those automatically-inserted hints.

Reply:
Sure. Accepted the suggestion.